### PR TITLE
Actions Interface

### PR DESCRIPTION
**Pull request by Glignos**
Implemented the actions interface based on the suggested mockups.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,14 +23,14 @@
     "@angular/platform-browser-dynamic": "^4.0.0",
     "@angular/router": "^4.0.0",
     "bootstrap": "^3.3.7",
-    "font-awesome": "^4.7.0",
     "core-js": "^2.4.1",
     "diff": "^3.3.0",
+    "font-awesome": "^4.7.0",
```
Comment (harunurhan) [inline package.json]: to `devDependencies`

File: src/app/action-template/action-template.component.html [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
+        <div *ngSwitchCase="'Update'">
+            <me-update-template [action]=action></me-update-template>
+        </div>
+        <div *ngSwitchCase="'Deletion'">
+            <me-deletion-template [action]=action></me-deletion-template>
+        </div>
+    </div>
+
+    <div *ngIf="canDelete" class="form-group">
+        <i class="fa fa-times fa-1" (click)="onRemoveClick()" aria-hidden="true"></i>
```
Comment (harunurhan) [inline src/app/action-template/action-template.component.html]: add bootstrap `btn` class if that changes too much, use `cursor: pointer` on `:hover`

File: src/app/actions-template/actions-template.component.html [hunk 3]
```diff
@@ -1,19 +1,42 @@
...
+    <option value="Deletion">Deletion</option>
+    <option value="Update">Update</option>
   </select>
 </div>
+<div class="bar">
+  Actions
+</div>
+<div class="info-text" [ngSwitch]="selectedAction">
+  <div *ngSwitchCase="'Addition'">
+    <p>Choose the field you'd like to add:</p>
```
Comment (harunurhan) [inline src/app/actions-template/actions-template.component.html]: `<span>`

File: src/app/actions-template/actions-template.component.scss [hunk 4]
```diff
@@ -1,20 +1,27 @@
...
     font-size: 12px;
     cursor: pointer;
     text-align: center;
     text-decoration: none;
     outline: none;
-    color: #FFFFFF;
-    background-color: #35BF16;
-    border: none;
-    border-radius: 15px;
+    color: rgb(0, 0, 0);
```
Comment (harunurhan) [inline src/app/actions-template/actions-template.component.scss]: `black`

File: src/app/actions-template/actions-template.component.ts [hunk 5]
```diff
@@ -6,7 +6,10 @@ import { AppGlobalsService } from '../shared/services';
 @Component({
   selector: 'me-actions-template',
   templateUrl: './actions-template.component.html',
-  styleUrls: ['./actions-template.component.scss']
+  styleUrls: [
+    './actions-template.component.scss',
+    '../multi-editor/multi-editor.component.scss'
```
Comment (harunurhan) [inline src/app/actions-template/actions-template.component.ts]: why this is here?, if you are using same classes put them to `styles.scss` not `multi-editor.component.scss`

File: src/app/multi-editor/multi-editor.component.ts [hunk 6]
```diff
@@ -12,7 +12,7 @@ import { UserActions } from '../shared/interfaces';
 @Component({
   selector: 'me-multi-editor',
   templateUrl: 'multi-editor.component.html',
-  styleUrls: ['multi-editor.component.scss'],
+  styleUrls: ['multi-editor.component.scss', '../app.component.scss'],
```
Comment (harunurhan) [inline src/app/multi-editor/multi-editor.component.ts]: ditto

File: src/app/update-template/update-template.component.html [hunk 7]
```diff
@@ -1,11 +1,19 @@
...
+    <input type="text" name="key" [(ngModel)]="action.mainKey" class="form-control input-sm" placeholder="field key...">
+  </div>
+  <span class="form-group">when value</span>
+  <div class="form-group">
+    <select class="form-control input-sm" name="matchType" [(ngModel)]="action.matchType">
+      <option *ngFor="let matchType of matchTypes" [ngValue]="matchType"> {{matchType}} </option>
+    </select>
+  </div>
+  <div class="form-group">
+    <input type="text" name="updateValue" [(ngModel)]="action.updateValue" class="form-control input-sm" placeholder="replace value...">
```
Comment (harunurhan) [inline src/app/update-template/update-template.component.html]: don't `...`

File: src/app/update-template/update-template.component.html [hunk 8]
```diff
@@ -1,11 +1,19 @@
...
+  </div>
+  <span class="form-group ">replace it with</span>
+  <div class="form-group">
+    <input type="text" name="value" [(ngModel)]="action.value" class="form-control input-sm" placeholder="new value..">
```
Comment (harunurhan) [inline src/app/update-template/update-template.component.html]: especially not `..`

File: src/app/update-template/update-template.component.scss [hunk 9]
```diff
@@ -1,56 +1,3 @@
...
-
-.regex {
-    margin-left:140px
-}
-
-.editor{
-    width: 600px;
-    height:400px;
+.selector{
+    width:200px;
```
Comment (harunurhan) [inline src/app/update-template/update-template.component.scss]: why this is hardcoded like this?

File: src/app/action-template/action-template.component.ts [hunk 10]
```diff
@@ -10,7 +10,7 @@ import { Action } from '../shared/interfaces';
 export class ActionTemplateComponent {
   @Input() id: number;
   @Input() action: Action;
-
+  @Input() canDelete: boolean;
```
Comment (harunurhan) [inline src/app/action-template/action-template.component.ts]: Better way:

- remove `canDelete`
- remove `id`
- remove `onActionDeleted`
- Move deletion button on parent

**Review by harunurhan — CHANGES_REQUESTED** [review]

File: src/app/action-template/action-template.component.html (hunk 2 above)
Comment (Glignos) [reply src/app/action-template/action-template.component.html]: yes the btn class makes the clicking area too big.Instead im using cursor:pointer and its working fine.Is there any need to be specifically on hover?

File: src/styles.scss [hunk 11]
```diff
@@ -1 +1,14 @@
 /* You can add global styles to this file, and also import other style files */
+html, body {
+    height: 100%;
+    background-color: #f9f9f9;
+  }
+
+
+.bar{
```
Comment (harunurhan) [inline src/styles.scss]: add a prefix `me-` since it's global style

File: src/app/conditions-template/conditions-template.component.ts [hunk 12]
```diff
@@ -10,6 +10,7 @@ import { AppGlobalsService } from '../shared/services';
 export class ConditionsTemplateComponent {
   @Input() id: number;
```
Comment (harunurhan) [inline src/app/conditions-template/conditions-template.component.ts]: remove

File: src/app/conditions-template/conditions-template.component.ts [hunk 13]
```diff
@@ -10,6 +10,7 @@ import { AppGlobalsService } from '../shared/services';
...
   @Input() condition: Condition;
+  @Input() canDelete: boolean;
```
Comment (harunurhan) [inline src/app/conditions-template/conditions-template.component.ts]: remove

File: src/app/conditions-template/conditions-template.component.ts [hunk 14]
```diff
@@ -10,6 +10,7 @@ import { AppGlobalsService } from '../shared/services';
...
   @Output() onConditionDeleted = new EventEmitter<number>();
```
Comment (harunurhan) [inline src/app/conditions-template/conditions-template.component.ts]: remove

File: src/app/actions-template/actions-template.component.ts [hunk 15]
```diff
@@ -48,15 +48,16 @@ export class ActionsTemplateComponent implements OnInit {
     this.onPreview.emit(this.userActions);
   }
 
-  onActionDeleted(id: number) {
+  onRemoveAction(id: number) {
```
Comment (harunurhan) [inline src/app/actions-template/actions-template.component.ts]: it's an index

**Review by harunurhan — CHANGES_REQUESTED** [review]
fix the git mess, then good to go

**Review by harunurhan — APPROVED** [review]

**Pull request merged**